File: Software/NodeJS/libs/commands.js
module.exports = {
  //Command Format
  // digitalRead() command format header
    dRead                                       : [1]
  // digitalWrite() command format header
  , dWrite                                      : [2]
  // analogRead() command format header
  , aRead                                       : [3]
  // analogWrite() command format header
  , aWrite                                      : [4]
  // pinMode() command format header
  , pMode                                       : [5]
  // Ultrasonic read
  , uRead                                       : [7]
  // Get firmware version
  , version                                     : [8]
  // Accelerometer (+/- 1.5g) read
  , acc_xyz                                     : [20]
  // RTC get time
  , rtc_getTime                                 : [30]
  // DHT Pro sensor temperature
  , dht_temp                                    : [40]

  // Grove LED Bar commands
  // Initialise
  , ledBarInit                                  : [50]
  // Set orientation
  , ledBarOrient                                : [51]
  // Set level
  , ledBarLevel                                 : [52]
  // Set single LED
  , ledBarSetOne                                : [53]
  // Toggle single LED
  , ledBarToggleOne                             : [54]
  // Set all LEDs
  , ledBarSet                                   : [55]
  // Get current state
  , ledBarGet                                   : [56]

  // Grove 4 Digit Display commands
  // Initialise
  , fourDigitInit                               : [70]
  // Set brightness, not visible until next cmd
  , fourDigitBrightness                         : [71]
  // Set numeric value without leading zeros
  , fourDigitValue                              : [72]
  // Set numeric value with leading zeros
  , fourDigitValueZeros                         : [73]
  // Set individual digit
  , fourDigitIndividualDigit                    : [74]
  // Set individual leds of a segment
  , fourDigitIndividualLeds                     : [75]
  // Set left and right values with colon
  , fourDigitScore                              : [76]
  // Analog read for n seconds
  , fourDigitAnalogRead                         : [77]
  // Entire display on
  , fourDigitAllOn                              : [78]
  // Entire display off
  , fourDigitAllOff                             : [79]

  // Grove Chainable RGB LED commands
  // Store color for later use
  , storeColor                                  : [90]
  // Initialise
  , chainableRgbLedInit                         : [91]
  // Initialise and test with a simple color
  , chainableRgbLedTest                         : [92]
  // Set one or more leds to the stored color by pattern
  , chainableRgbLedSetPattern                   : [93]
  // Set one or more leds to the stored color by modulo
  , chainableRgbLedSetModulo                    : [94]
  // Sets leds similar to a bar graph, reversible
  , chainableRgbLedSetLevel                     : [95]

  // Grove IR sensor
  // Read the button from IR sensor
  , irRead                                      : [21]
  // Set pin for the IR reciever
  , irRecvPin                                   : [22]

  // Grove Dust sensor
  , dustSensorRead                              : [10]
  , dustSensorEn                                : [14]
  , dustSensorDis                               : [15]

  // Encoder
  , encoderRead                                 : [11]
  , encoderEn                                   : [16]
  , encoderDis                                  : [17]

  // Grove Flow sensor
  , flowRead                                    : [12]
  , flowEn                                      : [18]
  , flowDis                                     : [13]

  // This allows us to be more specific about which commands contain unused bytes
  , unused                                      : 0
};

File: Software/Java/src/com/dexterind/grovepi/utils/Commands.java
package com.dexterind.grovepi.utils;

public class Commands {
  public static final int DREAD                           = 1;
  public static final int DWRITE                          = 2;
  public static final int AREAD                           = 3;
  public static final int AWRITE                          = 4;
  public static final int PMODE                           = 5;
  public static final int UREAD                           = 7;
  public static final int VERSION                         = 8;
  public static final int ACC_XYZ                         = 20;
  public static final int RTC_GET_TIME                    = 30;
  public static final int DHT_TEMP                        = 40;
  public static final int LEDBAR_INIT                     = 50;
  public static final int LEDBAR_ORIENT                   = 51;
  public static final int LEDBAR_LEVEL                    = 52;
  public static final int LEDBAR_SET_ONE                  = 53;
  public static final int LEDBAR_TOGGLE_ONE               = 54;
  public static final int LEDBAR_SET                      = 55;
  public static final int LEDBAR_GET                      = 56;
  public static final int FOUR_DIGIT_INIT                 = 70;
  public static final int FOUR_DIGIT_BRIGHTNESS           = 71;
  public static final int FOUR_DIGIT_VALUE                = 72;
  public static final int FOUR_DIGIT_VALUE_ZEROS          = 73;
  public static final int FOUR_DIGIT_INDIVIDUAL_DIGIT     = 74;
  public static final int FOUR_DIGIT_INDIVIDUAL_LEDS      = 75;
  public static final int FOUR_DIGIT_SCORE                = 76;
  public static final int FOUR_DIGIT_AREAD                = 77;
  public static final int FOUR_DIGIT_ALL_ON               = 78;
  public static final int FOUR_DIGIT_ALL_OFF              = 79;
  public static final int STORE_COLOR                     = 90;
  public static final int CHAINABLE_RGB_LED_INIT          = 91;
  public static final int CHAINABLE_RGB_LED_TEST          = 92;
  public static final int CHAINABLE_RGB_LED_SET_PATTERN   = 93;
  public static final int CHAINABLE_RGB_LED_SET_MODULO    = 94;
  public static final int CHAINABLE_RGB_LED_SET_LEVEL     = 95;
  public static final int IR_READ                         = 21;
  public static final int IR_RECV_PIN                     = 22;
  public static final int DUST_SENSOR_READ                = 10;
  public static final int DUST_SENSOR_EN                  = 14;
  public static final int DUST_SENSOR_DIS                 = 15;
  public static final int ENCODER_READ                    = 11;
  public static final int ENCODER_EN                      = 16;
  public static final int ENCODER_DIS                     = 17;
  public static final int FLOW_READ                       = 12;
  public static final int FLOW_EN                         = 18;
  public static final int FLOW_DIS                        = 13;

  public static final int UNUSED                          = 0;

  public Commands(){}
}